main/src/library/MutList.flix

```diff
@@ -32,6 +32,12 @@ instance Functor[MutList] {
     def map(f: a -> b & e, v: MutList[a]): MutList[b] & e = MutList.map(f, v) as & e
 }
 
+instance Copy[MutList[a]] {
+    def copy(v: MutList[a]): MutList[a] & e =
+        let MutList(a, l) = v;
+        MutList(ref (deref a), ref (deref l)) as & e
```

Comment:
This seems problematic to me because the underlying array is not copied. So this is not even a shallow copy.

I think we should do:

```
MutList(copy(a), ref (deref l))
```

This is still shallow (i.e. the elements of MutList are not copied) but the "MutList" structure is copied.

Does my comment make sense?

Reply:
It does :)